### PR TITLE
JDK-8300205: Swing test bug8078268 make latch timeout configurable

### PR DESCRIPTION
**Pull request by MBaesken**
The test javax/swing/text/html/parser/Parser/8078268/bug8078268.java has a latch timeout that is currently fix but should better be configurable to better deal with slow test infrastructures.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8300205](https://bugs.openjdk.org/browse/JDK-8300205): Swing test bug8078268 make latch timeout configurable

### Reviewers
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**) ⚠️ Review applies to [0c46dace](https://git.openjdk.org/jdk/pull/12011/files/0c46dacedc2992be45d04a627a5f27b649f66b99)
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**) ⚠️ Review applies to [0c46dace](https://git.openjdk.org/jdk/pull/12011/files/0c46dacedc2992be45d04a627a5f27b649f66b99)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12011/head:pull/12011` \
`$ git checkout pull/12011`

Update a local copy of the PR: \
`$ git checkout pull/12011` \
`$ git pull https://git.openjdk.org/jdk pull/12011/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12011`

View PR using the GUI difftool: \
`$ git pr show -t 12011`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12011.diff">https://git.openjdk.org/jdk/pull/12011.diff</a>

</details>

### CODE REVIEW

**Comment by mrserb** [comment]
I remember that match 5 uses timeout=4, so I wonder if timeout will be changed to 40 seconds will it be possible to verify old bug or not.

**Comment by aivanov-jdk** [comment]
> I remember that match 5 uses timeout=4, so I wonder if timeout will be changed to 40 seconds will it be possible to verify old bug or not.

The test still fails:
```text
./jdk-9-b123/bin/java -Dtest.timeout.factor=5 bug8078268
Exception in thread "main" java.lang.RuntimeException: Parsing takes too long. Current timeout is 50000
        at bug8078268.main(bug8078268.java:65)
```

It passes on jdk-9-b123 where [JDK-8078268](https://bugs.openjdk.org/browse/JDK-8078268) is fixed.

File: test/jdk/javax/swing/text/html/parser/Parser/8078268/bug8078268.java [hunk 1]
```diff
@@ -38,7 +38,8 @@
    @run main bug8078268
 */
 public class bug8078268 {
-    private static final long TIMEOUT = 10_000;
+    private static final float timeoutFactor = Float.parseFloat(System.getProperty("test.timeout.factor", "1.0"));
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/text/html/parser/Parser/8078268/bug8078268.java:41]:
```suggestion
    private static final float timeoutFactor =
            Float.parseFloat(System.getProperty("test.timeout.factor", "1.0"));
```
The line is 115 chars long, I suggest wrapping it. Using static import for `parseFloat` will also reduce the length; `getProperty` could also be statically imported.

**Review by mrserb — APPROVED** [review]

**Pull request closed without merging**